app/models/dossier.rb

```diff
@@ -670,6 +679,10 @@ def extend_conservation(conservation_extension)
       brouillon_close_to_expiration_notice_sent_at: nil,
       en_construction_close_to_expiration_notice_sent_at: nil,
       termine_close_to_expiration_notice_sent_at: nil)
+
+    if hidden_by_reason == 'expired'
+      update(hidden_by_administration_at: nil, hidden_by_user_at: nil, hidden_by_reason: nil)
```

Comment:
Voir plus bas sur le `hidden_by_expired_at`

```suggestion
      update(hidden_by_expired_at: nil, hidden_by_reason: nil)
```

un cas ou c'est nécessaire : 
- un utilisateur supprime sont dossier `hidden_by_user_at: today`
- plus tard, le dossier est soft deleted car proche de l'expiration `hidden_by_user_at = hidden_by_admin_at = later`
- l'administration veut finalement le conserver plus longtemps `hidden_by_user_at = hidden_by_admin_at = nil`
=> bug: le dossier réapparaît coté utilisateur

Reply:
et est-ce que pour éviter ça, mais sans créer une nouvelle colonne ce qui ne semblait pas plaire @tchak 
On pourrait faire qqchose comme
```
elsif author_is_automatic(author) && can_be_deleted_by_automatic?(reason)
   update(hidden_by_administration_at: Time.zone.now, hidden_by_user_at: hidden_by_user_at.presence || Time.zone.now, hidden_by_reason: reason)
```

et ici 
```
if hidden_by_reason == 'expired'
      if hidden_by_user_at !=  hidden_by_administration_at
        update(hidden_by_administration_at: nil, hidden_by_reason: nil)
      else
        update(hidden_by_administration_at: nil, hidden_by_user_at: nil, hidden_by_reason: nil)
      end
 end
```

Ou c'est trop tordu ? 😅

Reply:
Alors ca me semble compliqué et vu que la logique est déjà dur à comprendre ca me fait un poil peur d'en rajouter une couche. @tchak un avis ?

Reply:
Yes je suis d'accord ! Après dans les discussions qu'on avait eu avec Paul, il y a un petit moment maintenant, il me semble qu'on avait relevé "le bug" dont tu parles et on s'etait posé la question à quel cas d'usage il repondait dans la vie réelle. Et si c'etait pas un edge case, "qui au pire" faisait réapparaître un dossier masqué… Mais je laisse répondre Paul pour pas avoir l'air de ne vouloir plus retoucher à cette PR 😅